### PR TITLE
registers the .mlt file name extension and associate it with shotcut.exe

### PR DESCRIPTION
**Pull request by timlau**
I have added initial .mlt file assoiation to the windows NSIS installer

I don't have a windows system to test on at the moment, but is be easy to test and not break anything

### CODE REVIEW

File: packaging/windows/shotcut.nsi [hunk 1]
```diff
@@ -132,6 +141,8 @@ Section "Uninstall"
   ; Remove registry keys
   DeleteRegKey HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Shotcut"
   DeleteRegKey HKLM SOFTWARE\Shotcut
+  DeleteRegStr HKCR ".mlt"
+  DeleteRegStr HKCR "Shotcut\shell\open\command"
```
Comment (ddennedy) [inline packaging/windows/shotcut.nsi:145]: > Invalid command: "DeleteRegStr"

Should be `DeleteRegKey`

File: packaging/windows/shotcut.nsi [hunk 2]
```diff
@@ -101,6 +101,15 @@ Section "Create Start Menu Shortcut"
 
 SectionEnd
 
+; Optional section (can be disabled by the user)
+Section "Associate *.mlt files with Shotcut" 
+
+ ;register file extensions
+ WriteRegStr HKCR ".mlt" ""  "Shotcut"
+ WriteRegStr HKCR "Shotcut\shell\open\command" "" "$\"$INSTDIR\shotcut.exe$\" $\"%1$\""
```
Comment (ddennedy) [inline packaging/windows/shotcut.nsi:109]: change indentation to 2 spaces please

File: packaging/windows/shotcut.nsi (hunk 1 above)
Comment (ddennedy) [inline packaging/windows/shotcut.nsi:145]: Also, `DeleteRegStr HKCR "Shotcut\shell\open\command"`
should be `DeleteRegKey HKCR "Shotcut"` to delete the whole subtree we created.

**Review by ddennedy — CHANGES_REQUESTED** [review]
Thanks for your contribution! I can confirm it is working with some small changes.

**Comment by timlau** [comment]
I have solved the issues and made a minor change to the naming uses Shotcut.mlt instead of Shortcut, It is the naming style than most application uses. Ex. .7z uses 7Zip,7z

**Comment by ddennedy** [comment]
> change to the naming uses Shotcut.mlt 

OK, yes I see that too in registy, but then I thought how with the "Type" appear in Explorer? I do not want it to show "Shotcut.mlt". I looked in Explorer, and I see it is showing "MLT XML," but I cannot find that anywhere in the registry?! Where is it getting this. I needed to uninstall the Microsoft Store app to test this properly. Ah, I see in my AppxManifest.xml to make the store app, I register the file association and set the "DisplayName" to MLT XML along with a logo.

image


It must be caching that info somewhere. Well, I am not going to reboot just to see if that changes things.

**Pull request merged**